pkg/operation/botanist/component/vpa/exporter.go

```diff
@@ -0,0 +1,172 @@
+		service            = v.emptyService(exporter)
+		serviceAccount     = v.emptyServiceAccount(exporter)
+		clusterRole        = v.emptyClusterRole("exporter")
+		clusterRoleBinding = v.emptyClusterRoleBinding("exporter")
+		deployment         = v.emptyDeployment(exporter)
+		vpa                = v.emptyVerticalPodAutoscaler(exporter + "-vpa")
+	)
+
+	return resourceConfigs{
+		{obj: service, class: runtime, mutateFn: func() { v.reconcileExporterService(service) }},
+		{obj: deployment, class: runtime, mutateFn: func() { v.reconcileExporterDeployment(deployment, serviceAccount) }},
+		{obj: vpa, class: runtime, mutateFn: func() { v.reconcileExporterVPA(vpa, deployment) }},
+		{obj: serviceAccount, class: application, mutateFn: func() { v.reconcileExporterServiceAccount(serviceAccount) }},
+		{obj: clusterRole, class: application, mutateFn: func() { v.reconcileExporterClusterRole(clusterRole) }},
+		{obj: clusterRoleBinding, class: application, mutateFn: func() { v.reconcileExporterClusterRoleBinding(clusterRoleBinding, clusterRole, serviceAccount) }},
+	}
+}
+
+func (v *vpa) reconcileExporterService(service *corev1.Service) {
+	service.Labels = getAllLabels(exporter)
+	service.Spec = corev1.ServiceSpec{
+		Type:            corev1.ServiceTypeClusterIP,
+		SessionAffinity: corev1.ServiceAffinityNone,
+		Selector:        getAppLabel(exporter),
+	}
+
+	desiredPorts := []corev1.ServicePort{
+		{
+			Name:       "metrics",
+			Protocol:   corev1.ProtocolTCP,
+			Port:       exporterPortMetrics,
+			TargetPort: intstr.FromInt(int(exporterPortMetrics)),
+		},
+	}
+	service.Spec.Ports = kutil.ReconcileServicePorts(service.Spec.Ports, desiredPorts, corev1.ServiceTypeClusterIP)
+}
+
+func (v *vpa) reconcileExporterServiceAccount(serviceAccount *corev1.ServiceAccount) {
+	serviceAccount.AutomountServiceAccountToken = pointer.Bool(false)
+}
+
+func (v *vpa) reconcileExporterClusterRole(clusterRole *rbacv1.ClusterRole) {
+	clusterRole.Labels = getRoleLabel()
+	clusterRole.Rules = []rbacv1.PolicyRule{{
+		APIGroups: []string{"autoscaling.k8s.io"},
+		Resources: []string{"verticalpodautoscalers"},
+		Verbs:     []string{"get", "watch", "list"},
+	}}
+}
+
+func (v *vpa) reconcileExporterClusterRoleBinding(clusterRoleBinding *rbacv1.ClusterRoleBinding, clusterRole *rbacv1.ClusterRole, serviceAccount *corev1.ServiceAccount) {
+	clusterRoleBinding.Labels = getRoleLabel()
+	clusterRoleBinding.RoleRef = rbacv1.RoleRef{
+		APIGroup: rbacv1.GroupName,
+		Kind:     "ClusterRole",
+		Name:     clusterRole.Name,
+	}
+	clusterRoleBinding.Subjects = []rbacv1.Subject{{
+		Kind:      rbacv1.ServiceAccountKind,
+		Name:      serviceAccount.Name,
+		Namespace: serviceAccount.Namespace,
+	}}
+}
+
+func (v *vpa) reconcileExporterDeployment(deployment *appsv1.Deployment, serviceAccount *corev1.ServiceAccount) {
+	deployment.Labels = getAllLabels(exporter)
+	deployment.Spec = appsv1.DeploymentSpec{
+		Replicas:             pointer.Int32(1),
+		RevisionHistoryLimit: pointer.Int32(2),
```

Comment:
Nit: this was originally 1, not 2, here and in other deployments (not that it matters too much).

Reply:
Yes, we intentionally changed this for all refactored Helm charts because it helps checking why the pods of a `Deployment` got rolled (by comparing old and new `ReplicaSet`)